File: backend/src/main/java/com/eemall/demo/dto/Purchase.java
package com.eemall.demo.dto;

import com.eemall.demo.model.Order;
import com.eemall.demo.model.OrderItems;
import com.eemall.demo.model.customers.Address;
import com.eemall.demo.model.customers.Customer;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shipAddress;
    private Address billAddress;
    private Order order;
    private Set<OrderItems> orderItems;
}

File: frontend/src/app/common/purchase.ts

import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer: Customer;
    shipAddress: Address;
    billAddress: Address;
    order: Order;
    orderItems: OrderItem[]; 
}
